benches/decoding_benchmark.rs

```diff
@@ -0,0 +1,35 @@
+extern crate criterion;
+extern crate jpeg_decoder;
+
+
+use criterion::{black_box, Criterion, criterion_group, criterion_main};
+
+use jpeg_decoder as jpeg;
+use jpeg_decoder::ImageInfo;
+
+fn read_image(image: &[u8]) -> Vec<u8> {
+    jpeg::Decoder::new(black_box(image)).decode().unwrap()
+}
+
+fn read_metadata(image: &[u8]) -> ImageInfo {
+    let mut decoder = jpeg::Decoder::new(black_box(image));
+    decoder.read_info().unwrap();
+    decoder.info().unwrap()
+}
+
+pub fn criterion_benchmark(c: &mut Criterion) {
+    c.bench_function("decode a 512x512 image of colored peppers", |b| b.iter(|| {
+        read_image(include_bytes!("peppers.jpg"))
```

Comment:
`include_bytes!` does (as the name suggests) include the bytes as a reference to a byte array. I'm afraid the compiler can do optimizations because of this that aren't possible in most real usage of this library.

Reply:
No, the compiler doesn't do any optimization based on the contents of the byte array itself (you can look at the compiled output). 
It can't because of the `black_box` function we use.